Kuroba/app/src/main/java/com/github/adamantcheese/chan/report/ReportProblemPresenter.kt

```diff
@@ -0,0 +1,113 @@
+package com.github.adamantcheese.chan.report
+
+import android.os.Build
+import com.github.adamantcheese.chan.BuildConfig
+import com.github.adamantcheese.chan.Chan.inject
+import com.github.adamantcheese.chan.core.base.BasePresenter
+import com.github.adamantcheese.chan.core.base.MResult
+import com.github.adamantcheese.chan.utils.Logger
+import com.google.gson.Gson
+import com.google.gson.annotations.SerializedName
+import io.reactivex.Single
+import okhttp3.*
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.io.IOException
+import javax.inject.Inject
+
+class ReportProblemPresenter : BasePresenter<ReportProblemController>() {
+
+    @Inject
+    lateinit var okHttpClient: OkHttpClient
+    @Inject
+    lateinit var gson: Gson
+
+    override fun onCreate(view: ReportProblemController) {
+        super.onCreate(view)
+
+        inject(this)
+    }
+
+    fun sendReport(title: String, description: String, logs: String?): Single<MResult<Boolean>> {
+        require(title.isNotEmpty())
+        require(description.isNotEmpty())
```

Comment:
Just pointing this one out with that other comment about description being allowed to be empty, but logs have to be attached.

Reply:
Done.